docs/wallet/chain-query-api.md

```diff
@@ -0,0 +1,207 @@
+---
+id: chain-query-api
+title: chain-query-api
+---
+
+
+### ChainQueryApi
+
+The ChainQueryApi is how the wallet project stays aware of the current best chain.
+This allows the wallet for example to calculate the number of confirmations for a transaction,
+get the current chain tip, or even retrieve block filters for a given set of blocks.
+
+Since this is an API it can be hooked up to the `chain` module of bitcoin-s but it can also be linked to
+any other implementation of your choosing. This allows you to use the bitcoin-s wallet in any schema that you
+want.
+
+The functions that the ChainQueryApi supports are:
+
+```scala mdoc:compile-only
+import org.bitcoins.core.api.ChainQueryApi.FilterResponse
+import org.bitcoins.core.crypto.DoubleSha256DigestBE
+import org.bitcoins.core.protocol.BlockStamp
+
+import scala.concurrent.Future
```

Comment:
nit: Move these `import`s to an `invisible` block above

Reply:
You can merge these imports with the ones below (by bringing the ones below higher)